### PR TITLE
Adds recipients to Broadcasts

### PR DESCRIPTION
**Pull request by nicpottier**
We now have a recipients m2m that keeps track of all the URNs that a broadcast sent to. These are populated as messages are created so will be kept in sync.

This also takes care of backfilling all past broadcasts in a super mega migration that will need to be run outside a ship.

### CODE REVIEW

File: temba/msgs/migrations/0055_auto_20160427_1937.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    highpoint = r.get(HIGHPOINT_KEY)
+    if highpoint is None:
+        highpoint = 0
+
+    broadcast_ids = Broadcast.objects.filter(id__gt=highpoint).order_by('id').values_list('id', flat=True)
+    start = time.time()
+    for (i, broadcast_id) in enumerate(broadcast_ids):
+        recipient_count = populate_recipients_for_broadcast(Broadcast, MsgManager, broadcast_id)
+        print "%d - %d ... (%d of %d) in %d" % (broadcast_id, recipient_count, i, len(broadcast_ids)-1, int(time.time() - start))
+        r.set(HIGHPOINT_KEY, broadcast_id)
```
Comment (ericnewcomer) [inline temba/msgs/migrations/0055_auto_20160427_1937.py]: Have to be careful that this gets faked in the migration before running outside, or that the high point is manually cleared.

**Comment by ericnewcomer** [comment]
Looks good!

How long does each batch take in testing? Do you have a sense of how long it'll take to backfill?

File: temba/msgs/migrations/0055_auto_20160427_1937.py (hunk 1 above)
Comment (nicpottier) [inline temba/msgs/migrations/0055_auto_20160427_1937.py]: Ya, this needs faking for sure.

**Comment by nicpottier** [comment]
I don't actually have a super clear idea of how much time it'll take.. will know fairly quickly once I try though and reevaluate the approach if it looks overly ugly.

**Pull request merged**